pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/SumPrecisionAggregationFunction.java

```diff
@@ -77,40 +85,142 @@ public GroupByResultHolder createGroupByResultHolder(int initialCapacity, int ma
   @Override
   public void aggregate(int length, AggregationResultHolder aggregationResultHolder,
       Map<ExpressionContext, BlockValSet> blockValSetMap) {
-    byte[][] valueArray = blockValSetMap.get(_expression).getBytesValuesSV();
-    BigDecimal sumValue = getDefaultResult(aggregationResultHolder);
-    for (int i = 0; i < length; i++) {
-      BigDecimal value = DataTypeConversionFunctions.bytesToBigDecimalObject(valueArray[i]);
-      sumValue = sumValue.add(value);
+    BigDecimal sum = getDefaultResult(aggregationResultHolder);
+    BlockValSet blockValSet = blockValSetMap.get(_expression);
+    switch (blockValSet.getValueType()) {
+      case INT:
+        int[] intValues = blockValSet.getIntValuesSV();
+        for (int i = 0; i < length; i++) {
+          sum = sum.add(BigDecimal.valueOf(intValues[i]));
+        }
+        break;
+      case LONG:
+        long[] longValues = blockValSet.getLongValuesSV();
+        for (int i = 0; i < length; i++) {
+          sum = sum.add(BigDecimal.valueOf(longValues[i]));
+        }
+        break;
+      case FLOAT:
+      case DOUBLE:
```

Comment:
Float and double are treated as String ?

Reply:
We read them as string values. In order to construct `BigDecimal` from floating point values, we need to first convert the number into string format